programs/server/Server.cpp

```diff
@@ -1862,6 +1876,117 @@ void Server::createServers(
     http_params->setTimeout(settings.http_receive_timeout);
     http_params->setKeepAliveTimeout(keep_alive_timeout);
 
+    Poco::Util::AbstractConfiguration::Keys protocols;
```

Comment:
I still would ask you to move this code to a separate method/function. The method will not have more parameters than  `Server::createServers`, composable protocol creation code will be localized in certain method and will not blow up existing method.